src/cnd/cnd.ts

```diff
@@ -329,6 +329,218 @@ export interface SwapProperties {
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters for one side of the swap
+ */
+export interface Parameters {
+  // Do we want enums instead?
+  /**
+   * Protocol: "han", "herc20" or "halight"
+   */
+  protocol: string;
+  /**
+   * Ledger: "ethereum", "bitcoin" or "lightning"
+   */
+  ledger: string;
+  /**
+   * Network of the ledger: "mainnet", "testnet", "regtest" for Bitcoin, a chain id number for Ethereum
+   */
+  // This is tricky, an alternative would be to have more nesting where each ledger type (Bitcoin, Ethereum, Lightning) have a specific interface to declare network and asset
+  network: string | number;
+  /**
+   * Asset: "bitcoin", "ether" or { standard: "erc20", contract_address: "0x..." } for non-native tokens
+   */
+  // Not sure this is ideal, let's discuss and see what we can come up with
+  asset: string | { standard: "erc20"; contract_address: "0x..." };
+  /**
+   * Expiry: Expiry applying to this ledger, it can be an absolute or relative expiry, depending on the protocol and ledger
+   */
+  // We could have an object that allows the specification of the type of expiry here
+  expiry: number;
+}
+
+export enum EscrowStatus {
+  None = "NONE",
+  Initialized = "INITIALIZED",
+  Deployed = "DEPLOYED",
+  Funded = "FUNDED",
+  Redeemed = "REDEEMED",
+  Refunded = "REFUNDED",
+  IncorrectlyFunded = "INCORRECTLY_FUNDED"
+}
+
+/**
+ * The detailed description of the ledger state
+ */
+export interface LedgerState {
+  /**
+   * The address of the htlc.
+   */
+  htlc_location?: any;
+  /**
+   * The transaction ID of the deployment transaction.
+   */
+  deploy_tx?: string;
+  /**
+   * The transaction ID of the funding transaction.
+   */
+  fund_tx?: string;
+  /**
+   * The transaction ID of the redeem transaction.
+   */
+  redeem_tx?: string;
+  /**
+   * The transaction ID of the refund transaction.
+   */
+  refund_tx?: string;
+  /**
+   * The status of the escrow (HTLC or Invoice).
+   */
+  status: EscrowStatus;
+}
+
+// TODO: add `[k: string]: any;` in each section
+/**
+ * All the properties of a swap
+ */
+export interface FullSwapProperties {
+  /**
+   * The local id of the swap.
+   */
+  local_id: string;
+  /**
+   * The parameters of this swap.
+   */
+  // Decided to put everything together as now the protocols are tightly linked to the ledgers and assets
+  parameters: {
```

Comment:
I would be in favor of removing as much nesting as possible.

Reply:
I second the above point.